cacheable.decorator.ts

```diff
@@ -99,8 +99,8 @@ export function Cacheable(cacheConfig: IObservableCacheConfig = {}) {
                * if maxCacheCount has been passed, respect that and only shift the cachePairs if the new cachePair will make them exceed the count
                */
               if (
-                !cacheConfig.shouldCacheDecider ||
-                cacheConfig.shouldCacheDecider(response)
+                (!cacheConfig.shouldCacheDecider || cacheConfig.shouldCacheDecider(response)) &&
+                (!GlobalCacheConfig.shouldCacheDecider || GlobalCacheConfig.shouldCacheDecider(response))
```

Comment:
Thanks for the PR. I think this should be an OR condition since everywhere else we are giving priority to the local config and only then switch to the global one. While in your case, this will always execute both.

Reply:
I tried this but now some specs are failing in a way I resolve. Can you take a look?

Reply:
@angelnikolov I got stock in understanding the specs that are failing.  Can I persuade you to  take a look?
Thank you
Michael

Reply:
I will take a look when I find time :). Will ping you when I do.

Reply:
@krogh The tests fail in this case because if we have passed in a local `shouldCacheDecider` and it fails, we shouldn't go to the global cache decider which is what happens here.